src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs

```diff
@@ -296,6 +297,13 @@ public string PropertyName
         public abstract string TemplateHint { get; }
 
         /// <summary>
+        /// Gets an <see cref="IShouldValidate"/> implementation that indicates whether this model should be validated.
+        /// If <c>null</c>, properties with this <see cref="ModelMetadata"/> are validated.
+        /// </summary>
+        /// <value>Defaults to <c>null</c>.</value>
+        public abstract IShouldValidate ShouldValidate { get; }
```

Comment:
The analogues to `IPropertyFilterProvider` aren't very solid. For one thing, what would the `Attribute`s implement and how would users change the `Func<T, bool>`? Do you mean the `Attribute`s would implement `IPropertyValidationFilter` and that would contain a method which returns a method which returns a `bool`? If that's the case, where would `IPropertyValidationFilterProvider` be used that `Func<T, bool>` wouldn't be more appropriate?

For another, the names get very confusing unless we rename `IPropertyFilterProvider` to `IPropertyBindingFilterProvider`.

And, this feels like it adds a third (counting `ValidateChildren`) "primary" mechanism to provide static control of property validation. Not sure we need anything extra just for custom `IValidationMetadataProvider` support.

And, this feels like we're creating another layer of indirection to ease configuration in a custom `IValidationMetadataProvider`. We don't have a requirement for _two_ mechanisms here.

Let's take this offline&hellip;